From 50d2ca3e50f2e4fad4264d53921b131de7eb11d9 Mon Sep 17 00:00:00 2001
From: Alexander Graf <agraf@suse.de>
Date: Thu, 8 Jun 2017 09:07:23 +0200
Subject: [PATCH 135/174] arm64: Move voffset to .hyp.text
Patch-mainline: Never, only needed in 4.4 based kernel
References: bsc#1031492

The EL2 idmap code needs access to voffset now. Move it into it section, so it
automatically gets mapped.

Signed-off-by: Alexander Graf <agraf@suse.de>

---

I'm not sure why this patch is only needed in our SP3 code base - 4.12 seems
to live happily without it ...
---
 arch/arm64/mm/mmu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/mm/mmu.c b/arch/arm64/mm/mmu.c
index 92320fd..f5f2f58 100644
--- a/arch/arm64/mm/mmu.c
+++ b/arch/arm64/mm/mmu.c
@@ -48,7 +48,7 @@
 
 u64 idmap_t0sz = TCR_T0SZ(VA_BITS);
 
-u64 kimage_voffset __read_mostly;
+u64 kimage_voffset __section(.hyp.text);
 EXPORT_SYMBOL(kimage_voffset);
 
 /*
-- 
1.8.5.6

